src/extensions/lens-renderer-extension.ts

```diff
@@ -57,9 +57,12 @@ export class LensRendererExtension extends LensExtension {
   }
 
   /**
-   * Defines if extension is enabled for a given cluster. Defaults to `true`.
+   * Defines if extension is enabled for a given cluster. This method is only
+   * called when the extension is created within a cluster frame.
+   *
+   * The default implementation is to return `true`
    */
-  async isEnabledForCluster(cluster: Cluster): Promise<Boolean> {
+  async isEnabledForCluster(cluster: KubernetesCluster): Promise<Boolean> {
```

Comment:
Is this likely going to break some existing extensions? Can you overload (keep both the old and new) to avoid breaking?
https://stackoverflow.com/a/31815015
I guess the extension developer still has to override this version for it to work.

Reply:
The point is to not expose the `Cluster` type to the extension API. So we just have to accept this breakage. 

I have checked all the extensions listed on https://github.com/lensapp/lens-extensions and none of them use this feature so I think it is probably safe to do so.